tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -0,0 +1,275 @@
+    ]
+
+    @staticmethod
+    def get_dependency_map(extensions: List[Dict[str, Any]]) -> Dict[str, Dict[str, Any]]:
+        dependency_map: Dict[str, Dict[str, Any]] = dict()
+
+        for ext in extensions:
+            ext_name = ext['name']
+            provisioned_after = ext['properties'].get('provisionAfterExtensions')
+            depends_on = provisioned_after if provisioned_after else []
+            # We know an extension should fail if a script was provided
+            ext_settings = ext['properties'].get("settings")
+            should_fail = True if ext_settings and "script" in ext_settings else False
+            dependency_map[ext_name] = {"should_fail": should_fail, "depends_on": depends_on}
+
+        return dependency_map
+
+    @staticmethod
+    def get_sorted_extension_names(extensions: List[VirtualMachineScaleSetVMExtensionsSummary], ssh_client: SshClient) -> List[str]:
+        # Using VmExtensionIds to get publisher for each ext to be used in remote script
+        extension_full_names = {
+            "AzureMonitorLinuxAgent": VmExtensionIds.AzureMonitorLinuxAgent,
+            "RunCommandLinux": VmExtensionIds.RunCommand,
+            "CustomScript": VmExtensionIds.CustomScript
+        }
+        enabled_times = []
+        for ext in extensions:
+            # Only check extensions which succeeded provisioning
+            if "succeeded" in ext.statuses_summary[0].code:
+                enabled_time = ssh_client.run_command(f"ext_sequencing-get_ext_enable_time.py --ext_type {extension_full_names[ext.name]}",
+                                                      use_sudo=True)
+                enabled_times.append(
+                    {
+                        "name": ext.name,
+                        "enabled_time": datetime.strptime(enabled_time.replace('\n', ''), u'%Y-%m-%d %H:%M:%S')
+                     }
+                )
+
+        # sort the extensions based on their enabled datetime
+        sorted_extensions = sorted(enabled_times, key=lambda ext_: ext_["enabled_time"])
+        log.info("")
+        log.info("Extensions sorted by time they were enabled: {0}".format(
+            ', '.join(["{0}: {1}".format(ext["name"], ext["enabled_time"]) for ext in sorted_extensions])))
+        sorted_extension_names = [ext["name"] for ext in sorted_extensions]
+        return sorted_extension_names
+
+    @staticmethod
+    def validate_extension_sequencing(dependency_map: Dict[str, Dict[str, Any]], sorted_extension_names: List[str], relax_check: bool):
+        installed_ext = dict()
+
+        # Iterate through the extensions in the enabled order and validate if their depending extensions are already
+        # enabled prior to that.
+        for ext in sorted_extension_names:
+            # Check if the depending extension are already installed
+            if ext not in dependency_map:
+                fail("Unwanted extension found in VMSS Instance view: {0}".format(ext))
```

Comment:
this may fail on a shared vm if another test uses other extensions

Reply:
When we update the extensions profile in the scale set model, only extensions which are in our update should be on the VM. Any extensions on the VM which were not in the extensions profile would be disabled

Reply:
ah, yes, thanks

may be worth adding this as a comment (not that we sharing scale sets right now, but hey, prepare for the future!)